### DISCUSSION TITLE
Recently noticed throttled downloads

### DISCUSSION THREAD

**Discussion by Seeto** [ranaroussi/yfinance#2133] [Q&A]
Good morning  - since Wednesday or Thursday I've noticed significant throttling when downloading more than a few stocks. I understand this is a free source but I'm wondering if new limits have been imposed by yahoo or the library? Here's a program I'm running to log/diagnose my issue. I usually batch download some stock data with no issues, but here I'm attempting one at a time with different pauses between stocks. Any advice on navigating this? Thanks.

Edit: attempting to properly format code snippet. attached as 

[test_yfinance.txt](https://github.com/user-attachments/files/17791001/test_yfinance.txt)

[yfinance_test_5yr_debug_log.txt](https://github.com/user-attachments/files/17791002/yfinance_test_5yr_debug_log.txt)

**Comment by yunhongz** [comment]
Also find out, after 1000 stodk downloads, it is no longer available.

**Comment by ValueRaider** [comment]
Duplicate #2125
